core/src/main/java/com/linecorp/armeria/server/annotation/AdditionalHeader.java

```diff
@@ -16,12 +16,17 @@
 package com.linecorp.armeria.server.annotation;
 
 import java.lang.annotation.ElementType;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
 
 /**
  * Annotation for an additional HTTP header.
  */
-@Target(ElementType.ANNOTATION_TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+@Repeatable(AdditionalHeaders.class)
+@Target({ElementType.TYPE, ElementType.METHOD, ElementType.ANNOTATION_TYPE})
```

Comment:
nit: `ElementType.ANNOTATION_TYPE` is not necessary because `ElementType.TYPE` includes it.